NodeManager.cpp

```diff
@@ -1318,6 +1398,66 @@ void SensorBH1750::onReceive(const MyMessage & message) {
 #endif
 
 /*
+   SensorTSL2561
+*/
+#if MODULE_TSL2561 == 1
+// contructor
+SensorTSL2561::SensorTSL2561(int child_id): Sensor(child_id,A4) {
+  setPresentation(S_LIGHT_LEVEL);
+  setType(V_LEVEL);
+  _tsl = new TLS2561(TSL2561_ADDR_FLOAT);
+}
+
+// what do to during before
+void SensorTSL2561::onBefore() {
+  _tsl->begin();
+    // You can change the gain on the fly, to adapt to brighter/dimmer light situations
+  //tsl.setGain(TSL2561_GAIN_0X);         // set no gain (for bright situtations)
+  _tsl->setGain(TSL2561_GAIN_16X);      // set 16x gain (for dim situations)
+  // Changing the integration time gives you a longer time over which to sense light
+  // longer timelines are slower, but are good in very low light situtations!
+  _tsl->setTiming(TSL2561_INTEGRATIONTIME_13MS);  // shortest integration time (bright light)
+  //tsl.setTiming(TSL2561_INTEGRATIONTIME_101MS);  // medium integration time (medium light)
+  //tsl.setTiming(TSL2561_INTEGRATIONTIME_402MS);  // longest integration time (dim light)
+  
+}
+
+// what do to during setup
+void SensorTSL2561::onSetup() {
+  
+}
+
+// what do to during loop
+void SensorTSL2561::onLoop() {
+  // request the light level
+  _value_int = _tsl->getLuminosity(TSL2561_VISIBLE); 
+//  _value_int = _tsl->getLuminosity(TSL2561_FULLSPECTRUM); 
+//  _value_int = _tsl->getLuminosity(TSL2561_INFRARED); 
+  #if DEBUG == 1
+    Serial.print(F("BH1 I="));
+    Serial.print(_child_id);
+    Serial.print(F(" L="));
+    Serial.println(_value_int);
+  #endif
+  
+//  uint32_t lum = tsl.getFullLuminosity();
```

Comment:
No problem at all, I like the idea to add those new sensors very much, take you time! I probably need to open up a dev branch so when this will be ready it can be merged outside master